frontend/src/components/DateSelector.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from "react";
+
+export interface DateSelectorProps {
+  onChange: (value: Date) => void;
+  defaultDate: string | null | undefined;
+}
+
+const DateSelector: React.FC<DateSelectorProps> = ({
+  onChange,
+  defaultDate,
+}) => {
+  const [releaseDateTime, setReleaseDateTime] = useState<Date | null>(
+    new Date(),
+  );
+
+  useEffect(() => {
```

Comment:
No need for useEffect here.
Just use a function that calls onChange

Reply:
Lol yes, previous debugging step xDddd